### PR TITLE
1007 sighting re identify

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- New API to repeat sighting identification
- Returns the full Sighting data
- Full sighting data now containd idConfigs
- idConfigs can be patched in the same way as for AGS as requested

---

**REST API Updates **

```
[POST] /api/v1/sightings/{sighting_guid}/rerun_id/
{
}
```
Format of idConfigs retured in GET and to be used in PATCH: `[{'algorithms': ['hotspotter_nosv']}]`

### CODE REVIEW

**Comment by brmscheiner** [comment]
Three questions:
1. will this work if ID has not been run yet? for example if the user elects to skip running ID when they commit the sighting, or if the data is migrated. 
2. will this use the `idConfigs` property? Right now when kicking off ID during the commit step the `idConfigs` property must be patched onto the AGS. since we are dealing with sightings in this case, do i patch the same property on the sighting? and will the `idConfig` that was set on the AGS carry over to the sighting?
3. after kickoff, what will happen with the ID status properties @naknomum is setting up for 913?

**Comment by hwindsor** [comment]
> Three questions:
> 
> 1. will this work if ID has not been run yet? for example if the user elects to skip running ID when they commit the sighting, or if the data is migrated.
Yes

> 2. will this use the `idConfigs` property? Right now when kicking off ID during the commit step the `idConfigs` property must be patched onto the AGS. since we are dealing with sightings in this case, do i patch the same property on the sighting? and will the `idConfig` that was set on the AGS carry over to the sighting?
It will indeed use the idConfigs. The idConfigs are copied over from the AGS when the sighting is created.
There is not an api at the moment to patch the idConfigs on the Sighting. Do you want me to add that?  

> 3. after kickoff, what will happen with the ID status properties @naknomum is setting up for 913?
I'll leave that to @naknomum  to answer

**Comment by brmscheiner** [comment]
@hwindsor 
1. sounds great 
2. yes, definitely

**Comment by naknomum** [comment]
> Three questions:
> 
> 1. will this work if ID has not been run yet? for example if the user elects to skip running ID when they commit the sighting, or if the data is migrated.
> 2. will this use the `idConfigs` property? Right now when kicking off ID during the commit step the `idConfigs` property must be patched onto the AGS. since we are dealing with sightings in this case, do i patch the same property on the sighting? and will the `idConfig` that was set on the AGS carry over to the sighting?
> 3. after kickoff, what will happen with the ID status properties @naknomum is setting up for 913?

i am _just now_ starting to write `pipeline_status` for sightings (aka "for identification"), so i will keep this in mind.  the short answer is:  i will iterate over `jobs` _on sightings_ for that -- so it will largely depend on what jobs has in it.  :grin: 

as we discussed today ben, i plan to do a little `jobs: []` array anyway, so it should show up there.  i will also _attempt_ to use all the various jobs(??) to somehow summarize "where are we with this sighting?".  that, for sure, will be a work-in-progress as we run across edge-cases.  gulp.

**Review by drewblount — APPROVED** [review]
LGTM

**Comment by hwindsor** [comment]
> @hwindsor
> 
> 1. sounds great
> 2. yes, definitely. does it need to be a separate API or can the generic `PATCH /sighting/<guid>` support patching idConfigs?

@brmscheiner 
Not a separate API.
Addition of PATCH support of /idConfigs so yes exactly the same as for asset_groups

**Comment by hwindsor** [comment]
@bluemellophone @naknomum  All comments are now resolved and this is approved BUT probably worth giving @brmscheiner a chance to agree before merging.

**Comment by brmscheiner** [comment]
all sounds good to me, thanks!

**Review by brmscheiner — APPROVED** [review]

**Pull request merged**
